google-cloud-spanner/test/google/cloud/spanner/admin/instance/client_test.rb

```diff
@@ -0,0 +1,34 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "helper"
+require "google/cloud/spanner/admin/instance"
+require "google/cloud/spanner/admin/instance/v1/instance_admin"
+require "gapic/common"
+require "gapic/grpc"
+
+class Google::Cloud::Spanner::AdminInstanceClientConstructionMinitest < Minitest::Test
+  def test_instance_admin
+    emulator_host = "localhost:4567"
+    emulator_check = ->(name) { (name == "SPANNER_EMULATOR_HOST") ? emulator_host : nil }
+    # Clear all environment variables, except SPANNER_EMULATOR_HOST
+    ENV.stub :[], emulator_check do
+      Gapic::ServiceStub.stub :new, :stub do
+          client = Google::Cloud::Spanner::Admin::Instance.instance_admin project_id: "1234"
+          assert_kind_of Google::Cloud::Spanner::Admin::Instance::V1::InstanceAdmin::Client, client
+          assert_equal emulator_host, client.configure.endpoint
```

Comment:
In https://github.com/googleapis/google-cloud-ruby/pull/13456, I was also trying to assert if the credentials are `:this_channel_is_insecure`. But I couldn't get it working. Would you be able to figure out how to get it working?

Reply:
Thanks Shanika for pointing this out. The `credentials` is actually set to `:this_channel_is_insecure` at [L105](https://github.com/googleapis/google-cloud-ruby/pull/13456/files#diff-5a08f7db57758b2198fe4b1ebaa36dc9a71ba4fd082dfffa81486c2f8d889b72R105) but then it gets changed over [here](https://github.com/googleapis/google-cloud-ruby/pull/13456/files#diff-5a08f7db57758b2198fe4b1ebaa36dc9a71ba4fd082dfffa81486c2f8d889b72R121)
I'm not sure about the right behaviour here. Can you please help me out?

Reply:
@bajajneha27 @skuruppu Unfortunately the actual [`GRPC::Core::Channel`]() object does not expose the provided `credentials`, afaik. However, you can use `stub` to intercept and test the arguments provided to `GRPC::Core::Channel#new`. I will provide a GH suggestion.

Reply:
See suggested change, above.

Reply:
Thank you @quartzmo